drivers/display/display_otm8009a.c

```diff
@@ -0,0 +1,941 @@
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* not documented */
+	buf[0] = 0x40U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_NO_DOC1, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* power control settings 4 for DC voltage settings */
+	/* enable GVDD test mode */
+	buf[0] = 0x04U;
+	buf[1] = 0xA9U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_PWR_CTRL4, buf, 2);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* power control settings 2 for normal mode */
+	/* set pump 4 vgh voltage from 15.0v down to 13.0v */
+	/* set pump 5 vgh voltage from -12.0v downto -9.0v */
+	/* set pump 4&5 x6 (ONLY VALID when PUMP4_EN_ASDM_HV = "0") */
+	/* change pump4 clock ratio from 1 line to 1/2 line */
+	buf[0] = 0x96U;
+	buf[1] = 0x34U;
+	buf[2] = 0x01U;
+	buf[3] = 0x33U;
+	buf[4] = 0x33U;
+	buf[5] = 0x34U;
+	buf[6] = 0x33U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_PWR_CTRL2, buf, 7);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* panel driving mode */
+	/* set column inversion */
+	buf[0] = 0x50U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_P_DRV_M, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* VCOM voltage setting */
+	/* VCOM Voltage settings from -1.0000v downto -1.2625v */
+	buf[0] = 0x4EU;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_VCOMDC, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* oscillator adjustment for idle/normal mode */
+	/* set 65Hz */
+	buf[0] = 0x66U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_OSC_ADJ, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* RGB video mode setting */
+	buf[0] = 0x08U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_RGB_VID_SET, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* GVDD/NGVDD */
+	buf[0] = 0x79U;
+	buf[1] = 0x79U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_GVDDSET, buf, 2);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* source driver timing setting */
+	buf[0] = 0x0DU;
+	buf[1] = 0x1BU;
+	buf[2] = 0x02U;
+	buf[3] = 0x01U;
+	buf[4] = 0x3CU;
+	buf[5] = 0x08U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_SD_CTRL, buf, 6);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* panel type setting */
+	buf[0] = 0x00U;
+	buf[1] = 0x01U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_PANSET, buf, 2);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* GOA VST setting */
+	buf[0] = 0x85U;
+	buf[1] = 0x01U;
+	buf[2] = 0x00U;
+	buf[3] = 0x84U;
+	buf[4] = 0x01U;
+	buf[5] = 0x00U;
+	buf[6] = 0x81U;
+	buf[7] = 0x01U;
+	buf[8] = 0x28U;
+	buf[9] = 0x82U;
+	buf[10] = 0x01U;
+	buf[11] = 0x28U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_GOAVST, buf, 12);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* GOA CLKA1 setting */
+	buf[0] = 0x18U;
+	buf[1] = 0x04U;
+	buf[2] = 0x03U;
+	buf[3] = 0x39U;
+	buf[4] = 0x00U;
+	buf[5] = 0x00U;
+	buf[6] = 0x00U;
+	ret = otm8009a_mcs_write(dev, OTM8009A_MCS_GOACLKA1, buf, 7);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* GOA CLKA2 setting */
+	buf[0] = 0x18U;
+	buf[1] = 0x03U;
+	buf[2] = 0x03U;
+	buf[3] = 0x3AU;
+	buf[4] = 0x00U;
+	buf[5] = 0x00U;
+	buf[6] = 0x00U;
```

Comment:
note: I think some of these large "blobs" could be stored in ROM (`static const`).